### ISSUE TITLE
Setting the phases in the results tab should not reset selection in Dimensions

### ISSUE THREAD

**Issue by itsdani** [FTSRG/mondo-sam#6]
For example when looking at the results of only a limited set of tools, the Dimensions tab can be used to unselect the rest. However, any change in the Results tab (e.g. Phases or Metrics) resets the selection.

**Comment by zkovari** [comment]
Thank you, it seems like I missed out this feature in the Dimensions tab, I will fix this today.

**Comment by zkovari** [comment]
Fixed in [bccf837](https://github.com/FTSRG/mondo-sam/commit/bccf8375ce21d5b1cb1d3532dbb66f6f68ca2111).

**Comment by abelhegedus** [comment]
Thanks, two comments:
- `Al_l_ways select all` has a typo
- I think the default should be unselected

**Comment by zkovari** [comment]
Thanks, I didn't notice that.

By the way, I think the selected state by default is a better choice. Well, without giving any specific details about the operation of the MVC pattern which is found in the application, let me say that the unselected state would be more confusing especially for those users who are unfamiliar with the application. Additionally, I think that most of the users want to represent every possible data which belong to the particular legend.

**Comment by abelhegedus** [comment]
Without knowing the internal details, I would suggest the following:
- Initially (or when loading a results file or changing the "dimensions" settings) all dimensions should be selected.
- The check-box should read `Remember selection` and should be enabled by default. Disabling should cause the current behavior (reselect all on changes). If possible, add a tooltip or description somewhere on the page (or to the wiki/user guide at least).

**Comment by zkovari** [comment]
Basically, I agree with that this part needs some modifications. Considering your suggestions:
1. I agree and I guess this is happening right now.
2. The main problem about these selections in the Dimensions panel (these kind of specific legend filters) is that their values depend on the current legend and the selected states of the widgets in the Results panel as well such as Scenario, Case, Size, and the like. Which means, if the `Remember selection` was enabled (or the currently used `Always Select All` was disabled) by default, we could lose some data during the visualisation, without even changing the dimensions panel. Just imagine a situation when different values (specific legends, for example `L1`, `L2`, `L3`, `L4`) belong to various scenarios (`S1` and `S2`). `L1`, `L2`, `L3` belong to `S1`, and `L2`, `L4` belong to `S2`. Assume that, the first chosen scenario is `S1`, therefore, we are able to visualise `L1`, `L2` and `L3`. However, since the `Remember selection` is enabled by default, by changing the scenario from `S1` to `S2`, we lose `L4` in the plots, and only `L2` can be depicted (because it was in the previous selections, amongst `L1`, `L2` and `L3`). Actually, this was the problem that I called confusing in my previous comment, I just didn't want to bother you with the details. 

Anyway, this `Remember selection` sounds good (but being disabled by default), and I agree with that a little guide could be useful.

**Comment by abelhegedus** [comment]
I see. Also, is there something that makes selecting multiple scenarios problematic?

In the CPS example, we use two scenarios batch and incremental but would like to visualize results from phases that are common to both.

**Comment by zkovari** [comment]
Well, it causes some changes in the optimization process, and probably some performance loss will occur. But it is absolutely feasible.

**Comment by zkovari** [comment]
Sorry, what is the goal exactly? To summarise the measurement values of various scenarios, or to compare the results of them on the same diagrams? The first requires the introduction of multiple selections of scenarios like in the case of metrics and phases, and the latter necessitates to represent the scenario as a dimension. For example, choosing scenario as legend would solve the problem. But I guess you would like to visualise size on the x-axis, cases (queries) as legends, and somehow also show the differences between the two scenarios. In this case, we need [facets](http://www.cookbook-r.com/Graphs/Facets_%28ggplot2%29/).

**Comment by abelhegedus** [comment]
Compare the [batch](https://github.com/IncQueryLabs/incquery-examples-cps/blob/master/tests/org.eclipse.incquery.examples.cps.performance.tests/src/org/eclipse/incquery/examples/cps/performance/tests/config/scenarios/ToolChainPerformanceBatchScenario.xtend) and [incremental](https://github.com/IncQueryLabs/incquery-examples-cps/blob/master/tests/org.eclipse.incquery.examples.cps.performance.tests/src/org/eclipse/incquery/examples/cps/performance/tests/config/scenarios/ToolChainPerformanceIncrementalScenario.xtend) scenarios in CPS.

I would like to compare the M2MTransformation phases where some tools are run with the batch scenario, some with the incremental.

Note, that maybe it would be better to have a single scenario instead of two...

**Comment by zkovari** [comment]
Yeah, I see. Well, until you do not want visualise other dimensions such as sizes or cases, it is feasible by choosing Tool on the x-axis, and Scenario as Legend. Otherwise, I should implement facets in the application, which actually would be a useful feature.
